src/planner.rs

```diff
@@ -92,6 +93,113 @@ impl<C> Drop for SystemGuard<C> {
     }
 }
 
+/// A builder for the `Planner` struct.
+/// All `with_*` methods return `self` to allow chained calls.
+/// 
+/// ## World
+///
+/// You have to specify a world, otherwise the
+/// `build` method will panic.
+///
+/// ## Number of threads
+///
+/// Specifying the number of threads to use
+/// for the thread pool is optional.
+/// 
+/// **Note:** Only specify this if you do not
+/// want to share the thread pool, because then
+/// the number of threads the pool is configured with
+/// will determince this property.
+///
+/// # The thread pool
+///
+/// Also optional, use this if you want to
+/// share the thread pool with other parts of
+/// your crate (you should do this if you use
+/// threads somwhere else).
+///
+/// If you didn't specify a thread pool nor a number of
+/// threads, a new thread pool with the number of virtual
+/// cpus will be created.
+pub struct PlannerBuilder {
+    world: Option<World>,
```

Comment:
Yes, I thought about using `with_thread_pool`, but I thought it's not descriptive enough because you still have another parameter. Yes, you are right I could have asked for world already in the builder constructor, but I think it is nicer to use. Of course it's not compile-time enforced then.